engine/gamesys/src/gamesys/resources/res_font.cpp

```diff
@@ -0,0 +1,426 @@
+        DeleteFontResource(params->m_Factory, font_map);
+        return dmResource::RESULT_OK;
+    }
+
+    dmResource::Result ResFontRecreate(const dmResource::ResourceRecreateParams* params)
+    {
+        FontResource* font_map = (FontResource*)dmResource::GetResource(params->m_Resource);
+
+        dmRenderDDF::FontMap* ddf;
+        dmDDF::Result e = dmDDF::LoadMessage<dmRenderDDF::FontMap>(params->m_Buffer, params->m_BufferSize, &ddf);
+        if ( e != dmDDF::RESULT_OK )
+        {
+            return dmResource::RESULT_FORMAT_ERROR;
+        }
+
+        dmResource::Result r = AcquireResources(params->m_Factory, (dmRender::HRenderContext) params->m_Context, ddf, font_map, params->m_Filename);
+        if(r != dmResource::RESULT_OK)
+        {
+            return r;
+        }
+
+        dmResource::SetResourceSize(params->m_Resource, GetResourceSize(font_map));
+        return dmResource::RESULT_OK;
+    }
+
+    dmRender::HFont ResFontGetHandle(FontResource* resource)
+    {
+        return resource->m_FontMap;
+    }
+
+    dmResource::Result ResFontGetInfo(FontResource* font, FontInfo* desc)
+    {
+        memcpy(desc, font->m_DDF, sizeof(FontInfo));
+        return dmResource::RESULT_OK;
+    }
+
+    dmResource::Result ResFontSetCacheCellSize(FontResource* font, uint32_t cell_width, uint32_t cell_height, uint32_t max_ascent)
+    {
+        dmRender::SetFontMapCacheSize(font->m_FontMap, cell_width, cell_height, max_ascent);
+        return dmResource::RESULT_OK;
+    }
+
+    dmResource::Result ResFontGetCacheCellSize(FontResource* font, uint32_t* cell_width, uint32_t* cell_height, uint32_t* max_ascent)
+    {
+        dmRender::GetFontMapCacheSize(font->m_FontMap, cell_width, cell_height, max_ascent);
+        return dmResource::RESULT_OK;
+    }
+
+    bool ResFontHasGlyph(FontResource* font, uint32_t codepoint)
+    {
+        return 0 != font->m_DynamicGlyphs.Get(codepoint) || 0 != font->m_Glyphs.Get(codepoint);
+    }
+
+    dmResource::Result ResFontAddGlyph(FontResource* font, uint32_t codepoint, FontGlyph* inglyph, void* imagedata, uint32_t imagedatasize)
+    {
+        if (font->m_DynamicGlyphs.Full())
+        {
+            uint32_t cap = font->m_DynamicGlyphs.Capacity() + 64;
+            font->m_DynamicGlyphs.SetCapacity((cap*3)/2, cap);
+        }
+
+        DynamicGlyph* glyph = new DynamicGlyph;
+        // dmRender::Glyph is currently a dmRenderDDF::GlyphBank::Glyph
+        dmRenderDDF::GlyphBank::Glyph& g = glyph->m_Glyph;
+        g.m_Character  = codepoint;
+        g.m_Width      = inglyph->m_Width;
+        g.m_Advance    = inglyph->m_Advance;
+        g.m_LeftBearing= inglyph->m_LeftBearing;
+        g.m_Ascent     = inglyph->m_Ascent;
+        g.m_Descent    = fabs(inglyph->m_Descent);
+        // Redundant in this setup
+        // g.m_X;
+        // g.m_Y;
+        // g.m_GlyphDataOffset;
+        // g.m_GlyphDataSize;
```

Comment:
Are these not used anymore, or why are they redundant?

Reply:
the X/Y was never used at runtime. only in the editor.

And, the dynamic system doesn't produce the data in the same way, so the glyph data is different and handled explicitly.